clarity/src/vm/database/clarity_db.rs

```diff
@@ -915,6 +927,31 @@ impl<'a> ClarityDatabase<'a> {
         }
     }
 
+    pub fn get_block_height_for_tenure_height(
+        &mut self,
+        tenure_height: u32,
+    ) -> Result<Option<u32>> {
+        let current_tenure_height = self.get_tenure_height()?;
+        if current_tenure_height < tenure_height {
+            return Ok(None);
+        }
+        let current_height = self.get_current_block_height();
+        if current_height <= tenure_height {
```

Comment:
In Stacks 2.x, block height and tenure height are the same.  If `current_height == tenure_height`, and we're in epoch 2.x, then wouldn't this be `Ok(Some(tenure_height))`?

Reply:
Returning `None` is okay in that case, because the re-interpreted tenure height couldn't be evaluated in that case anyways (`get-block-info` must short-circuit to`None` rather than attempting to evaluate at the current block height).

I can add a rustdoc comment to this function to describe this behavior, but figuring out whether or not `tenure_height` is in `2.x` without querying information about the current block is hard and would end overcomplicating this method.

Reply:
Sure, a rustdoc would suffice

Reply:
Added in https://github.com/stacks-network/stacks-core/pull/5321/commits/d901503496a4c91671f9700a0e4df972beb1aafd